shotover/src/server.rs

```diff
@@ -277,6 +285,39 @@ impl<C: CodecBuilder + 'static> TcpCodecListener<C> {
             backoff *= 2;
         }
     }
+
+    /// Handle hot reload request by extracting file descriptor and responding
+    async fn handle_hot_reload_request(&mut self, request: HotReloadListenerRequest) {
+        let response = if let Some(ref listener) = self.listener {
+            // Extract the file descriptor from the TcpListener
+            let fd = listener.as_raw_fd();
+
+            // Always parse port from listen_addr to support hostnames and [IPv6]:port formats.
```

Comment:
The word `Always` only makes sense knowing the context that the previous implementation had a fallback.

The previous comment was something like: `split once from the right to support hostnames and [IPv6]:port formats.`
Mentioning that splitting once from the right lets us support both of these cases is valuable IMO, so lets keep that original comment.